frontend/src/utils/date.ts

```diff
@@ -0,0 +1,35 @@
+import dayjs from "dayjs"
+import timezone from "dayjs/plugin/timezone"
+import utc from "dayjs/plugin/utc"
+import { isoDateTimePattern } from "./validation"
+
+/**
+ * Format given date to specified format based relative to user's timezone.
```

Comment:
do i have to format the input to use this function? or can i pass in any date and it's already formatted according to the ISO 8601?

Reply:
The input date needs to be in ISO 8601 format. This function was intended to format the dateTime field from our postgreSQL tables.

Reply:
ah ok.  maybe make that more clear in the function title.  that this is for formatting dates from the database.  right now, i am wondering:  
1.  dayjs gives us a general format function -- what is the special use case for this specialized function
2.  how do i format the date to pass into the formatDate funciton?

It makes more sense now that you explain this is most useful for database conversions

Reply:
Renamed `formatDate` to `formatISODateToUserTimezone` and updated function doc:

```typescript
/**
 * Format given date to specified format based relative to user's timezone.
 * Use this function to format the dateTime field retrieved from the database.
 * @param date - The date string to format (expected to be in ISO 8601 date format).
 * @returns Formatted date string or "Invalid Date" if the input is not valid.
 *
 * Examples:
 *  Input: "2024-09-23T12:00:00Z"
 *  Expected Output: "Sep 23, 2024 08:00AM" (if user's in America/New_York timezone)
 */
export const formatISODateToUserTimezone = (date: string): string => {
  const errorMessage = "Invalid Date"

  if (typeof date !== "string" || !isoDateTimePattern.value.test(date))
    return errorMessage
  // Unsafe access to 'error' typed value is handled by dayjs' isValid function
  /* eslint-disable */
  const parsedDate = dayjs.utc(date)
  const userTimezone = dayjs.tz.guess()
  return parsedDate.isValid()
    ? parsedDate.tz(userTimezone).format("MMM DD, YYYY hh:mm A")
    : errorMessage
  /* eslint-enable */
}

```